PetaPoco/Providers/OracleDatabaseProvider.cs

```diff
@@ -35,6 +35,23 @@ public override void PreExecute(IDbCommand cmd)
         {
             cmd.GetType().GetProperty("BindByName")?.SetValue(cmd, true, null);
             cmd.GetType().GetProperty("InitialLONGFetchSize")?.SetValue(cmd, -1, null);
+            
+            if (cmd.CommandType == CommandType.StoredProcedure)
+            {
+                //Oracle stored procedure parameter names do not use the parameter prefix
+                //They also need to match the db exactly, so it's up to the consumer to specify the correct name
+                //In Database.AddParameter, the parameter prefix is prepended and it didn't seem right to cater for this case there...
+                //...so we need to undo those changes here
+                var paramPrefix = GetParameterPrefix(null);
+                var enumerator = cmd.Parameters.GetEnumerator();
+                while (enumerator.MoveNext())
+                {
+                    var parameter = (IDataParameter)enumerator.Current;
+
+                    if (parameter.ParameterName.StartsWith(paramPrefix, StringComparison.OrdinalIgnoreCase))
+                        parameter.ParameterName = parameter.ParameterName.Substring(paramPrefix.Length);
+                }
+            }
```

Comment:
What if providers had a `Dictionary<CommandType, string>` that the `Database` can use to figure out which parameter prefix to use? Then this code would not be necessary. The `OracleDatabaseProvider` uses `null` or `string.Empty` for `StoredProcedure` type and ":" for everything else.

Reply:
I'm inclined to agree with you; not incredibly excited about relying on an "undo" operation unless there's absolutely no other option.

I personally don't see much of an issue with handling it in `AddParameter`; it's only one additional branch, and that only under certain conditions...something like this (untested; both added lines are commented)?

```cs
private void AddParameter(IDbCommand cmd, object value, PocoColumn pc)
{
    //...

    // Support passed in parameters
    if (value is IDbDataParameter idbParam)
    {
        if (cmd.CommandType == CommandType.Text)
            idbParam.ParameterName = cmd.Parameters.Count.EnsureParamPrefix(_paramPrefix);
        else if (idbParam.ParameterName?.StartsWith(_paramPrefix) != true)
        {
            // only add param prefix if it's not an Oracle stored procedures
            if (!(cmd.CommandType == CommandType.StoredProcedure && _provider is Providers.OracleDatabaseProvider))
                idbParam.ParameterName = idbParam.ParameterName.EnsureParamPrefix(_paramPrefix);
        }
        cmd.Parameters.Add(idbParam);
    }
    else
    {
        var p = cmd.CreateParameter();
        // only add param prefix if it's not an Oracle stored procedures
        if (!(cmd.CommandType == CommandType.StoredProcedure && _provider is Providers.OracleDatabaseProvider))
            p.ParameterName = cmd.Parameters.Count.EnsureParamPrefix(_paramPrefix);
        SetParameterProperties(p, value, pc);

        cmd.Parameters.Add(p);
    }
}
```

Reply:
Absolutely love it! It's exactly what I was looking for. Can I include it in this PR after confirming everything still works?

Reply:
Sure, go for it.

Reply:
Done.